Lh 3624

rubene opened the pull request:
Added newrelic deployment hook

extra parameters can be sent:
deployment[app_name]            The name of the application, found in the newrelic.yml file
deployment[application_id]  The ID # of the application
deployment[description]     Text annotation for the deployment — notes
deployment[revision]            The revision number from your source control system (SVN, git, etc.)
deployment[changelog]           A list of changes for this deployment
deployment[user]                    The name of the user/process that triggered this deployment

In order for the Hook to work you have to add the :newrelic recipe to the user_recipes on the Capfiles of the projects.

This was tested using my_ama pointing the gem to the github branch
    gem 'capistrano-deploy', github: 'amaabca/capistrano-deploy', branch: 'LH-3624'
there is a branch on my_ama (LH-3624) that you can use for testing.

rubene committed:
Add newrelic hook

rubene committed:
fix api key

ryanjones commented:
Let's do this:

deployment[app_name] The name of the application, found in the newrelic.yml file
deployment[revision] The revision number from your source control system (SVN, git, etc.)
deployment[user] The name of the user/process that triggered this deployment

Let's disable the campfire hook for now.

rubene committed:
Change app name

* Change app name for a generic one,
will capture the app name being deploy.

rubene committed:
Add revision variable

* Add revision variable. will grab the last
commit id fron the log.
* Move the api key to be an ENV variable.
* Refactor notice deployment body to use variables
instead of strings.

rubene committed:
Add user variable

* Add user vairable using the current
user logged in on terminal.
* Change app name to user the env one
instead of the one in cap file. the one
in cap file differs from the one on new relic
giving an error on deploy.

rubene committed:
Change vairable names

* change variable names for app_name
and user for it to not conflict with
the ones on capfile.

rubene committed:
Try new way to get the current user

rubene commented:
The following data was added to the deployment hook: deployment[app_name], deployment[revision], deployment[user]. In addition change the api-key to be an env variable.

In order for this to work on a project:
-The newrelic recipe have to be added to the cap file
-EVNs have to be update to add the following variables:
NEW_RELIC_APP_NAME (this have to match the app name on new relic)
NEW_RELIC_API_KEY

jbabe commented on lib/capistrano-deploy/newrelic.rb:
```diff
@@ -0,0 +1,24 @@
+    def self.load_into(configuration)
+      configuration.load do
+
+        set(:api_key) { ENV['NEW_RELIC_API_KEY'] }
```
I'd move this into the Capfile:

```
set(:new_relic_key), '1111111111111'
set(:new_relic_app_name), 'foobar'
```

Reading directly from the `ENV` will cause a capistrano failure if `cap` is run without foreman or dotenv

jbabe commented on lib/capistrano-deploy/newrelic.rb:
```diff
@@ -0,0 +1,24 @@
+          end
+        end
+
+        after 'unicorn:reexec', 'newrelic:notice_deployment'
```
I might move this into the Capfile as well - this adds a dependancy on someone using `unicorn:reexec` - which might not always be used.

rubene committed:
Move new relic vairables outside the recipe

* Move the app name and api key to be on the
capfile.
* Change the user to get the git user instead
of the command line user.

MathieuGilbert commented:
If the key needs to be in the ENV files, we should get some PRs in for them before merging this in.
:elephant: :elephant: :elephant:

rubene committed:
Add description to deployment hook

* Add description variable that will
hold the stage of the project on deploy.

rubene commented:
Added stage of deployment to the description

```
 deployment[description]='#{current_stage}'
```

rubene committed:
Fix typo

rubene commented:
@MathieuGilbert Good catch

MathieuGilbert commented:
:elephant: :elephant: :elephant: :gun:

rubene committed:
Remove the stage variable

* New relic only tracks deployment
events on production.

MathieuGilbert commented:
:elephant: :elephant: :elephant:
